builtin/logical/pki/path_manage_issuers.go

```diff
@@ -318,3 +320,167 @@ either take PEM-formatted certificates, and, if :type="bundle", unencrypted
+	}
+
+	// If its already been revoked, just return the read results sans warnings
+	// like we would otherwise.
+	if issuer.Revoked {
+		return respondReadIssuer(issuer)
+	}
+
+	// When revoking, we want to forbid new certificate issuance. We allow
+	// new revocations of leaves issued by this issuer to trigger a CRL
+	// rebuild still.
+	issuer.Revoked = true
+	if issuer.Usage.HasUsage(IssuanceUsage) {
+		issuer.Usage.ToggleUsage(IssuanceUsage)
+	}
+
+	currTime := time.Now()
+	issuer.RevocationTime = currTime.Unix()
+	issuer.RevocationTimeUTC = currTime.UTC()
+
+	err = sc.writeIssuer(issuer)
+	if err != nil {
+		return nil, err
+	}
+
+	// Rebuild the CRL to include the newly revoked issuer.
+	crlErr := b.crlBuilder.rebuild(ctx, b, req, false)
+	if crlErr != nil {
+		switch crlErr.(type) {
+		case errutil.UserError:
+			return logical.ErrorResponse(fmt.Sprintf("Error during CRL building: %s", crlErr)), nil
+		default:
+			return nil, fmt.Errorf("error encountered during CRL building: %w", crlErr)
+		}
+	}
+
+	// Finally, respond with the issuer's updated data.
+	response, err := respondReadIssuer(issuer)
+	if err != nil {
+		// Impossible.
+		return nil, err
+	}
+
+	// For sanity, we'll add a warning message here if there's no other
+	// issuer which verifies this issuer.
+	ourCert, err := issuer.GetCertificate()
+	if err != nil {
+		return nil, err
+	}
+
+	allIssuers, err := sc.listIssuers()
+	if err != nil {
+		return nil, err
+	}
+
+	isSelfSigned := false
+	haveOtherIssuer := false
+	for _, candidateID := range allIssuers {
+		candidate, err := sc.fetchIssuerById(candidateID)
+		if err != nil {
+			return nil, err
+		}
+
+		candidateCert, err := candidate.GetCertificate()
+		if err != nil {
+			// Returning this error is fine because more things will fail
+			// if this issuer can't parse.
+			return nil, err
+		}
+
+		if err := ourCert.CheckSignatureFrom(candidateCert); err == nil {
+			// Signature verification is a success. This means we have a
+			// parent for this cert. But notice above we didn't filter out
+			// ourselves: we want to see if this is a self-signed cert. So
+			// check that now.
+			if candidate.ID == issuer.ID {
+				isSelfSigned = true
+			} else {
+				haveOtherIssuer = true
+			}
+		}
+
+		// If we have both possible warning candidates, no sense continuing
+		// to check signatures; exit.
+		if isSelfSigned && haveOtherIssuer {
+			break
+		}
+	}
+
+	if isSelfSigned {
+		response.AddWarning("This issuer is a self-signed (potentially root) certificate. This means it may not be considered revoked if there is not an external, cross-signed variant of this certificate. This issuer's serial number will not appear on its own CRL.")
+	}
+
+	if !haveOtherIssuer {
+		response.AddWarning("This issuer lacks another parent issuer within the mount. This means it will not appear on any other CRLs and may not be considered revoked by clients. Consider adding this issuer to its issuer's CRL as well if it is not self-signed.")
+	}
+
```

Comment:
Ah, great point about the default. 


Just as a note, from a philosophy perspective, this isn't really about a _certificate_ itself being revoked (the revoke action), this is more of an accounting trick, similar to soft deletion. The actual revocation needs to occur at the parent's mount, which may or may not be this mount point. If this is the same mount, then this does take care of that revocation.

But since the `pathFetchValid` handler needs to know about intermediate revocation, we should write that storage entry you talk about above :-)